src/test/scala/org/apache/spark/sql/delta/MergeIntoSuiteBase.scala

```diff
@@ -1878,4 +1880,79 @@ abstract class MergeIntoSuiteBase
       (1, 10) // update
     )
   )
+
+  protected def withJobCountListener(jobCount: AtomicInteger)(f: => Unit): Unit = {
+    val listener = new SparkListener {
+      override def onJobStart(jobStart: SparkListenerJobStart): Unit = {
+        // Spark will always log a job start/end event even when the job does not launch any task.
+        if (jobStart.stageInfos.exists(_.numTasks > 0)) {
+          jobCount.incrementAndGet()
+        }
+      }
+    }
+    sparkContext.addSparkListener(listener)
+    sparkContext.listenerBus.waitUntilEmpty(15000)
+    jobCount.set(0)
+
+    Utils.tryWithSafeFinally(f) {
+      sparkContext.listenerBus.waitUntilEmpty(15000)
+      sparkContext.removeSparkListener(listener)
+    }
+  }
+
+  test("findTouchedFiles - no files match target predicate") {
+
+    withTable("source") {
+      append(Seq((1, 10), (2, 20)).toDF("key1", "value1"), "value1")  // target
+      val source = Seq((3, 30)).toDF("key2", "value2")  // source
+
+      val jobCount = new AtomicInteger(0)
+
+      withJobCountListener(jobCount) {
+        io.delta.tables.DeltaTable.forPath(spark, tempPath)
```

Comment:
Please look at other tests in this test suite and see how they execute the merge using this method "executeMerge". Using that pattern would help in the future to run the same test on SQL as well.